Nondeterminism in text_plugin_test

https://github.com/tensorflow/tensorboard/pull/1021 introduced a test regression, likely caused by the hash map dict.keys() method introducing nondeterministic order.

```
======================================================================
FAIL: testPluginTagsImpl (__main__.TextPluginTest)
testPluginTagsImpl (__main__.TextPluginTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tensorboard/plugins/text/text_plugin_test.py", line 404, in testPluginTagsImpl
    }, self.plugin.tags_impl())
AssertionError: {'fry': ['message', 'vector'], 'leela': ['message', 'vector']} != {'fry': [u'vector', u'message'], 'leela': [u'vector', u'message']}
repr() of differing entries:
'fry': ['message', 'vector'] != [u'vector', u'message']
'leela': ['message', 'vector'] != [u'vector', u'message']
```

I'm going to workaround this internally. This test should be fixed.

However I suspect this might be the tip of the iceberg with the multiplexer and accumulator. I'd very much like for us to move in a direction where our serving endpoints are deterministic, and not impacted by underlying hashing algorithms. SQL should help.

--- a/tensorboard/plugins/text/text_plugin_test.py
+++ b/tensorboard/plugins/text/text_plugin_test.py
@@ -397,19 +397,19 @@ class TextPluginTest(tf.test.TestCase):
 
     # Initially, the thread for checking for plugin assets data has not run.
     # Hence, the mapping should only have data from the multiplexer.
-    self.assertEqual({
-        'fry': ['message', 'vector'],
-        'leela': ['message', 'vector']
-    }, self.plugin.tags_impl())
+    run_to_tags = self.plugin.tags_impl()
+    self.assertItemsEqual(['fry', 'leela'], run_to_tags.keys())
+    self.assertItemsEqual(['message', 'vector'], run_to_tags['fry'])
+    self.assertItemsEqual(['message', 'vector'], run_to_tags['leela'])
     thread = self.plugin._index_impl_thread
     mock.assert_called_once_with(thread)
 
     # The thread hasn't run yet, so no change in response, and we should not
     # have tried to launch a second thread.
-    self.assertEqual({
-        'fry': ['message', 'vector'],
-        'leela': ['message', 'vector']
-    }, self.plugin.tags_impl())
+    run_to_tags = self.plugin.tags_impl()
+    self.assertItemsEqual(['fry', 'leela'], run_to_tags.keys())
+    self.assertItemsEqual(['message', 'vector'], run_to_tags['fry'])
+    self.assertItemsEqual(['message', 'vector'], run_to_tags['leela'])
     mock.assert_called_once_with(thread)
 
     # Run the thread; it should clean up after itself.

